<!--Respected instructors please run the nodes separately in different terminals along with launch file .First scripts/qr.py ur5_2 then scripts/play.py ur5_1-->
 


<launch>
    

    <!-- Launch Task-5 Simulation Environment in Gazebo -->
    <include file="$(find pkg_vb_sim)/launch/task5_simulation.launch" />

    <node pkg="pkg_ros_iot_bridge" type="node_action_server_ros_iot_bridge.py" name="node_action_server_ros_iot_bridge" 
    output="screen"/>
 
    <node pkg="pkg_task5" type="shell_webpages_launch.sh" name="shell_webpages_launch" output="screen">
        <param name="cmd" value="$ pkg_task5/launch/shell_webpages_launch.sh"/>
    </node>

    <rosparam file ="$(find pkg_ros_iot_bridge)/config/config_iot.yaml"/> 
   

  <group ns="ur5_1">
    <!-- <rosparam command="load" file="$(find pkg_moveit_ur5)/config/joint_names.yaml" /> -->

    <include file="$(find pkg_moveit_ur5_1_config)/launch/planning_context.launch">
      <arg name="load_robot_description" value="true"/>
    </include>

    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
      <param name="/use_gui" value="false"/>
      <rosparam param="/source_list">[/joint_states]</rosparam>
    </node>
     
    

    <include file="$(find pkg_moveit_ur5_1_config)/launch/move_group.launch">
      <arg name="publish_monitored_planning_scene" value="true"/>
    </include>
    <include file="$(find pkg_moveit_ur5_1_config)/launch/moveit_rviz_robot1.launch">
      <arg name="config" value="true"/>
    </include>

    <!-- Add Shelf to Moveit! Planning Scene in RViz -->
    <arg name="scene_file" default="$(find pkg_task5)/config/ur5d.scene"/> 
    
    <node name = "moveit_publish_scene_from_text" pkg= "moveit_ros_planning" type = "moveit_publish_scene_from_text" args= "$(arg scene_file)"/>


  </group>


  <group ns="ur5_2">
    <!-- <rosparam command="load" file="$(find pkg_moveit_ur5)/config/joint_names.yaml" /> -->

    <include file="$(find pkg_moveit_ur5_2_config)/launch/planning_context.launch">
      <arg name="load_robot_description" value="true"/>
    </include>

    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
      <param name="/use_gui" value="false"/>
      <rosparam param="/source_list">[/joint_states]</rosparam>
    </node>

    <include file="$(find pkg_moveit_ur5_2_config)/launch/move_group.launch">
      <arg name="publish_monitored_planning_scene" value="true"/>
    </include>
    <include file="$(find pkg_moveit_ur5_2_config)/launch/moveit_rviz_robot2.launch">
      <arg name="config" value="true"/>
    </include>

    <!-- Add Shelf to Moveit! Planning Scene in RViz -->
    <arg name="scene_file" default="$(find pkg_task5)/config/new8.scene"/> 
    
    <node name = "moveit_publish_scene_from_text" pkg= "moveit_ros_planning" type = "moveit_publish_scene_from_text" args= "$(arg scene_file)"/>


  </group>


    <!-- Run the Online Order Placer Node -->
    <!--rosparam file ="$(find pkg_task4)/config/config_online_order.yaml"/-->
    <!--node name= "node_online_order_placer" pkg= "pkg_vb_sim" type="node_online_order_placer.py" output="screen"/-->


    

    
   

    <!-- Recording Bag File for Submission -->
    <arg name="record" default="false"/>
    <arg name="rec_name" default="task5.bag"/>

    <group if="$(arg record)">
        <node name="rosbag_record_pick" pkg="rosbag" type="record"
         args="record -O $(find pkg_task5)/bag_files/$(arg rec_name) --chunksize=10 /eyrc/vb/ur5_1/vacuum_gripper/logical_camera/ur5_1"       output="screen"/>
   </group>

</launch>

